File: Frontend/src/api/audit.ts
import { apiRequest } from "./client";

export type AuditLog = {
    id: string
    action: string
    entity_type: string
    entity_id: string
    summary: string
    actor: string
    details: Record<string, unknown>
    occurred_at: string
}

export function fetchAuditLogs(limit = 500): Promise<AuditLog[]>{
    const query = new URLSearchParams({
        limit: String(limit),
    })
    return apiRequest<AuditLog[]>(`/audit?${query.toString()}`)
}

File: Backend/app/models/audit.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class AuditLog(BaseModel):
    id: str
    action: str
    entity_type: str
    entity_id: str
    summary: str
    actor: str = "local_operator"
    details: dict[str, Any] = Field(default_factory=dict)
    occurred_at: datetime